examples/nlp/question_answering/question_answering_squad_infer.py

```diff
@@ -0,0 +1,300 @@
+    parser.add_argument(
+        "--do_lower_case",
+        action='store_true',
+        help="Whether to lower case the input text. True for uncased models, False for cased models.",
+    )
+    parser.add_argument(
+        "--no_data_cache", action='store_true', help="When specified do not load and store cache preprocessed data.",
+    )
+    parser.add_argument(
+        "--doc_stride",
+        default=128,
+        type=int,
+        help="When splitting up a long document into chunks, how much stride to take between chunks.",
+    )
+    parser.add_argument(
+        "--max_query_length",
+        default=64,
+        type=int,
+        help="The maximum number of tokens for the question. "
+        "Questions longer than this will be truncated to "
+        "this length.",
+    )
+    parser.add_argument(
+        "--max_seq_length",
+        default=384,
+        type=int,
+        help="The maximum total input sequence length after "
+        "WordPiece tokenization. Sequences longer than this "
+        "will be truncated, and sequences shorter than this "
+        " will be padded.",
+    )
+    parser.add_argument("--num_gpus", default=1, type=int, help="Number of GPUs")
+    parser.add_argument(
+        "--amp_opt_level", default="O0", type=str, choices=["O0", "O1", "O2"], help="01/02 to enable mixed precision"
+    )
+    parser.add_argument("--local_rank", type=int, default=None, help="For distributed training: local_rank")
+    parser.add_argument(
+        "--work_dir",
+        default='output_squad',
+        type=str,
+        help="The output directory where the model predictions and checkpoints will be written.",
+    )
+    parser.add_argument(
+        "--version_2_with_negative",
+        action="store_true",
+        help="If true, the SQuAD examples contain some that do not have an answer.",
+    )
+    parser.add_argument(
+        '--null_score_diff_threshold',
+        type=float,
+        default=0.0,
+        help="If null_score - best_non_null is greater than the threshold predict null.",
+    )
+    parser.add_argument(
+        "--n_best_size",
+        default=20,
+        type=int,
+        help="The total number of n-best predictions to generate in the nbest_predictions.json output file.",
+    )
+    parser.add_argument("--batches_per_step", default=1, type=int, help="Number of iterations per step.")
+    parser.add_argument(
+        "--max_answer_length",
+        default=30,
+        type=int,
+        help="The maximum length of an answer that can be "
+        "generated. This is needed because the start "
+        "and end predictions are not conditioned "
+        "on one another.",
+    )
+    parser.add_argument(
+        "--output_prediction_file",
+        type=str,
+        required=False,
+        default="predictions.json",
+        help="File to write predictions to. Only in evaluation mode.",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def create_pipeline(
```

Comment:
if you're reusing the same pipeline, would it be simpler to add inference to the existing example?

Reply:
makes sense